File: src/main/java/org/example/websitesalephone/service/user/UserService.java
package org.example.websitesalephone.service.user;

import org.example.websitesalephone.comon.CommonResponse;
import org.example.websitesalephone.dto.user.CreateUserDto;
import org.example.websitesalephone.dto.user.ProfileUserRequest;
import org.example.websitesalephone.dto.user.UserSearchForm;

public interface UserService {

    CommonResponse getUserByLoginId(String loginId);

    CommonResponse createUser(CreateUserDto userDto);

    CommonResponse updateUser(CreateUserDto userDto);

    CommonResponse deleteUser(String userId);

    CommonResponse search(UserSearchForm searchForm);

    CommonResponse updateProfileUser(ProfileUserRequest profileUserRequest, String id);

    CommonResponse getUserById(String id);
}

File: cms-user/src/service/UserService.ts
import type { AxiosResponse } from 'axios'
import { CreateUserDto, type ICreateUserDto } from '../models/CreateUserDto'
import type { UserSearchForm } from '../models/UserSearchForm'
import type { ProfileUserRequest } from '../models/ProfileUserRequest'
import api from '../api/api'

class UserService {
    private readonly ROOT_API = `${import.meta.env.VITE_ROOT_API}/api/user/`

    public getUserByLoginId(loginId: string): Promise<AxiosResponse> {
        return api.get(`${this.ROOT_API}get/user/${loginId}`)
    }

    public createUser(createUserDto: CreateUserDto): Promise<AxiosResponse> {
        return api.post(`${this.ROOT_API}create`, createUserDto.toPayload())
    }

    public updateUser(updateUserDto: ICreateUserDto): Promise<AxiosResponse> {
        return api.put(`${this.ROOT_API}update`, updateUserDto)
    }

    public deleteUser(userId: string): Promise<AxiosResponse> {
        return api.put(`${this.ROOT_API}delete/${userId}`)
    }

    public search(searchForm: UserSearchForm): Promise<AxiosResponse> {
        return api.post(`${this.ROOT_API}search`, searchForm.toPayload())
    }

    public updateProfileUser(profileUserRequest: ProfileUserRequest): Promise<AxiosResponse> {
        return api.put(`${this.ROOT_API}update-profile-user`, profileUserRequest)
    }
}

export const userService = new UserService()
